### PR TITLE
randomize pd test container name 

### LINKED ISSUE #26141: Fix PD tests and move out of flaky
PR #https://github.com/kubernetes/kubernetes/pull/26089 moved some PD tests to flaky to resolve issue https://github.com/kubernetes/kubernetes/issues/26127 (and duplicate #26076). These tests started failing due to PR #21709.

@rootfs is working on a patch to de-flake the tests.

This issue is to track the fix and should be closed when:

1. The PD tests have been deflaked (verify by running the "pd.go" e2e tests back to back for a couple of hours to verify the issue is fixed).
2. The flaky label has been removed from the "pd.go" tests that were marked flaky in https://github.com/kubernetes/kubernetes/pull/26089

CC @kubernetes/sig-storage

### PR DESCRIPTION
**Pull request by rootfs**
fix #26141
based on [comments](https://github.com/kubernetes/kubernetes/pull/26100#issuecomment-221103398) from @ncdc

### CODE REVIEW

File: test/e2e/pd.go [hunk 1]
```diff
@@ -431,9 +427,10 @@ func detachPD(hostName, pdName string) error {
 func testPDPod(diskNames []string, targetHost string, readOnly bool, numContainers int) *api.Pod {
 	containers := make([]api.Container, numContainers)
 	for i := range containers {
-		containers[i].Name = "mycontainer"
+		name := "mycontainer" + strconv.Itoa(mathrand.Int())
```
Comment (fejta) [inline test/e2e/pd.go]: A monotonically increasing number unique to the program might work better here.

**Comment by fejta** [comment]
LGTM with some comments. @saad-ali do you want to take a look?

**Comment by saad-ali** [comment]
LGTM

@rootfs did you run the tests to verify they are working. The merge bot doesn't run these tests.

**Comment by rootfs** [comment]
@saad-ali yes i ran tests overnight.

**Comment by saad-ali** [comment]
Taking another look at this PR. It actually won't change anything since the pod/container names are still generated once per test. Every iteration creating and tearing down the same pod will still end up with the same pod/container name. Am I missing something?

**Comment by rootfs** [comment]
@saad-ali before this change, the container name is always 'mycontainer'. This change appends a random number to the container name like the following

```console
Jun  9 16:55:27.298: INFO: Running '/srv/dev/kubernetes/_output/local/bin/linux/amd64/kubectl exec --namespace=e2e-tests-pod-disks-lh723 pd-test-e8c5f20d-2e62-11e6-a1ff-b8ca3a62792c -c=mycontainer36928506542153397011 -- cat /testpd1/tracker2'
```

**Comment by ncdc** [comment]
From an exec standpoint, it's the pod name that needs to be unique, not the container name. It looks like that's already unique per pod (https://github.com/rootfs/kubernetes/blob/acfcbffd42ae92242c0b949f938152b3814de9c6/test/e2e/pd.go#L457)?

**Comment by rootfs** [comment]
thanks @ncdc for the insight. 

The Pod name is created before the test starts, so even it is unique between tests, it is still deterministic in the same test https://github.com/rootfs/kubernetes/blob/acfcbffd42ae92242c0b949f938152b3814de9c6/test/e2e/pd.go#L180

**Comment by ncdc** [comment]
Ok, yeah, I didn't have a chance to read through the entire test file to find that. So yes, you need to create a new pod for each test case and test iteration, especially if you're deleting a pod and recreating.

**Comment by rootfs** [comment]
Let me check the failed the tests again.

**Comment by rootfs** [comment]
@ncdc @saad-ali the following log [from](https://00e9e64baccb46054ec7a4d395bb32e26b714116f2c8853cc5-apidata.googleusercontent.com/download/storage/v1_internal/b/kubernetes-jenkins/o/logs%2Fkubernetes-e2e-gce-flaky%2F18504%2Fbuild-log.txt?qk=AD5uMEtcW-u2jQydTR9wW-QdKqgsDARc4QX_3jMHIwdIDWmGRFdiAYxlifz_2lY-r2-81Mn3rOgRgkpwU5Kh7-wfK88S_fA_ASZcJCpxqqAM-HAASKIU2ufsxOKzb7G6pTvVk7a0IgB7stMF2K2AARn-I7JVRiG08UCQof_RHAr4oWgD_Hi9HNfmaFGBcAr_mtM-WO9estibTCTNf3cIX4h_99r1XUsQ0KvGw4wu9lMBK8uyL19Glo1XBuhsFXpUTdEnexU6hDUmLNRKbXmJOVnVV2Nd0wypYw4EMpvrpNmxzX6ndQz14FQOng-Mwcb0nV-0vA4BelisYXa8Igd_c6N35v9Sxg37PTw_8Ariz3FL_I3A04lcN4blUG9OHYR1IuFYtF3vIFtGySDZEPxuFgRoFmbJg_oJanh8Cd2neEAbfXM8sUdNj8WdRERHa9_obhKVolyumTtYI9R2SApya5BLPXYEgGeYp7skKJ_giBISkd1OXPqL32WVvj6w-FRWYPrexr9lc1SS-AZSetnD4Gq_3qCZiOdvnKVCQgxUL4uGWptq8UHWOSaqLhPbsomYoSbATuAdJWD-O2Eu0ObayDh3mhb24QsPYZhiR_fen2rQCW8bGJik093Ap3oJCc5vlwVK2LShaMai9GNhNGFPd5U2yLgIRp9rCDzMVbH042DppmAOia5cNVw3GoMKZRmRBMJR0kO_4eH_v30msndfqeZz_XTkNPart7Udq6Vh9q-WBEmw1nmCRIJh6G9F0Ft-tNq-NDpBYq2VI6pRcYrdXAx5Em-VZLrYDf1ZhJanRL55lr9zuAzFpBeaC5GCxZV8IK83KJOazEK3Fgr17osELK_VCXYlknACcvHMAlRgJDnXMhWZS8mNA73zG9yFyzgS8Wf38aR2Vje4)  doesn't convince me this fix can help. .In the failure, the container names were different (mycontainer3 and mycontainer4) and yet still failed

```text
[k8s.io] Pod Disks 
  should schedule a pod w/ a RW PD shared between multiple containers, write to PD, delete pod, verify contents, and repeat in rapid succession [Slow] [Flaky]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/pd.go:218
[BeforeEach] [k8s.io] Pod Disks
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:120
STEP: Creating a kubernetes client
Jun 10 00:31:50.961: INFO: >>> kubeConfig: /workspace/.kube/config

STEP: Building a namespace api object
STEP: Waiting for a default service account to be provisioned in namespace
[BeforeEach] [k8s.io] Pod Disks
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/pd.go:69
[It] should schedule a pod w/ a RW PD shared between multiple containers, write to PD, delete pod, verify contents, and repeat in rapid succession [Slow] [Flaky]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/pd.go:218
STEP: creating PD
Jun 10 00:31:54.923: INFO: Successfully created a new PD: "jenkins-e2e-65b25155-2edd-11e6-8aff-0242ac11000a".
Jun 10 00:31:54.923: INFO: PD Read/Writer Iteration #0
STEP: submitting host0Pod to kubernetes
W0610 00:31:54.935102    8734 request.go:347] Field selector: v1 - pods - metadata.name - pd-test-6805af1f-2edd-11e6-8aff-0242ac11000a: need to check if this is versioned correctly.
STEP: writing a file in the container
Jun 10 00:32:04.921: INFO: Running '/workspace/kubernetes/platforms/linux/amd64/kubectl exec --namespace=e2e-tests-pod-disks-8bfcu pd-test-6805af1f-2edd-11e6-8aff-0242ac11000a -c=mycontainer3 -- /bin/sh -c echo '7396730354903675317' > '/testpd1/tracker0''
Jun 10 00:32:05.261: INFO: Wrote value: "7396730354903675317" to PD "jenkins-e2e-65b25155-2edd-11e6-8aff-0242ac11000a" from pod "pd-test-6805af1f-2edd-11e6-8aff-0242ac11000a" container "mycontainer3"
STEP: reading a file in the container
Jun 10 00:32:05.261: INFO: Running '/workspace/kubernetes/platforms/linux/amd64/kubectl exec --namespace=e2e-tests-pod-disks-8bfcu pd-test-6805af1f-2edd-11e6-8aff-0242ac11000a -c=mycontainer4 -- cat /testpd1/tracker0'
Jun 10 00:32:05.487: INFO: Read file "/testpd1/tracker0" with content: 7396730354903675317

STEP: deleting host0Pod
Jun 10 00:32:05.525: INFO: PD Read/Writer Iteration #1
STEP: submitting host0Pod to kubernetes
W0610 00:32:05.532123    8734 request.go:347] Field selector: v1 - pods - metadata.name - pd-test-6805af1f-2edd-11e6-8aff-0242ac11000a: need to check if this is versioned correctly.
STEP: reading a file in the container
Jun 10 00:32:07.055: INFO: Running '/workspace/kubernetes/platforms/linux/amd64/kubectl exec --namespace=e2e-tests-pod-disks-8bfcu pd-test-6805af1f-2edd-11e6-8aff-0242ac11000a -c=mycontainer1 -- cat /testpd1/tracker0'
Jun 10 00:32:07.272: INFO: Read file "/testpd1/tracker0" with content: 7396730354903675317

STEP: writing a file in the container
Jun 10 00:32:07.273: INFO: Running '/workspace/kubernetes/platforms/linux/amd64/kubectl exec --namespace=e2e-tests-pod-disks-8bfcu pd-test-6805af1f-2edd-11e6-8aff-0242ac11000a -c=mycontainer3 -- /bin/sh -c echo '2662366901720958409' > '/testpd1/tracker1''
Jun 10 00:32:07.509: INFO: Wrote value: "2662366901720958409" to PD "jenkins-e2e-65b25155-2edd-11e6-8aff-0242ac11000a" from pod "pd-test-6805af1f-2edd-11e6-8aff-0242ac11000a" container "mycontainer3"
STEP: reading a file in the container
Jun 10 00:32:07.509: INFO: Running '/workspace/kubernetes/platforms/linux/amd64/kubectl exec --namespace=e2e-tests-pod-disks-8bfcu pd-test-6805af1f-2edd-11e6-8aff-0242ac11000a -c=mycontainer4 -- cat /testpd1/tracker0'
Jun 10 00:32:07.732: INFO: Read file "/testpd1/tracker0" with content: 7396730354903675317

STEP: reading a file in the container
Jun 10 00:32:07.733: INFO: Running '/workspace/kubernetes/platforms/linux/amd64/kubectl exec --namespace=e2e-tests-pod-disks-8bfcu pd-test-6805af1f-2edd-11e6-8aff-0242ac11000a -c=mycontainer4 -- cat /testpd1/tracker1'
Jun 10 00:32:07.876: INFO: error running kubectl exec to read file: exit status 1
stdout=rpc error: code = 2 desc = "oci runtime error: exec failed: lstat /proc/10488/ns/ipc: no such file or directory"
stderr=error: error executing remote command: error executing command in container: Error executing in Docker Container: 126
)
Jun 10 00:32:07.876: INFO: Error reading file: exit status 1
Jun 10 00:32:07.876: INFO: Unexpected error occurred: exit status 1
STEP: cleaning up PD-RW test environment
STEP: Waiting for PD "jenkins-e2e-65b25155-2edd-11e6-8aff-0242ac11000a" to detach from "jenkins-e2e-minion-group-9hts"
Jun 10 00:32:14.756: INFO: GCE PD "jenkins-e2e-65b25155-2edd-11e6-8aff-0242ac11000a" appears to have successfully detached from "jenkins-e2e-minion-group-9hts".
STEP: Deleting PD "jenkins-e2e-65b25155-2edd-11e6-8aff-0242ac11000a"
Jun 10 00:32:18.587: INFO: Successfully deleted PD "jenkins-e2e-65b25155-2edd-11e6-8aff-0242ac11000a".
```

**Comment by ncdc** [comment]
Correct, this won't help. You can't do this reliably:

1. Create pod abcd
1. Delete pod abcd
1. Create pod abcd
1. `kubectl exec abcd -- <commands>`

The problem is that you have 2 pods with the same name but different UUIDs, and exec doesn't support UUIDs right now, so it picks the "wrong" (deleted) pod from steps 1/2.

**Pull request closed without merging**